### PR TITLE
ostree: ignore exit code 65 for systemd-tmpfiles

### PR DESCRIPTION
**Pull request by gicmo**
When pre-filling /var after deploying the ostree commit, do not interpret return code 65 by systemd-tmpfiles as an error. This return code is used to indicate that some of the lines in the configuration were ignore, which could be due to a syntax error but also something as harmless as duplicated entries. Either way it the contents of `/lib/tmpfiles.d` is provided the content and thus out of the hand of the installer itself. Since there was no error executing the not-ignored lines we should not abort the installation.

### CODE REVIEW

**Comment by poncovka** [comment]
Please, report a bug in bugzilla and add `Resolves: rhbz#<bug-number>` at the end of the commit  message.

**Comment by gicmo** [comment]
Sorry about that, added the `Resolves:` line.

File: pyanaconda/payload/rpmostreepayload.py [hunk 1]
```diff
@@ -392,11 +392,26 @@ def _prepare_mount_targets(self):
...
+            # According to systemd-tmpfiles(8), the return values are:
+            #  0 → success
+            # 65 → so some lines had to be ignored, but no other errors
+            # 73 → configuration ok, but could not be created
+            #  1 → other error
+            # Therefore we ignore error 65, since this is coming from
+            # the payload itself and the actual execution of it was fine
+            if rc not in [0, 65]:
+                exn = PayloadInstallError(
+                    "%s %s exited with code %d" % (cmd, argv, rc))
```
Comment (VladimirSlavik) [inline pyanaconda/payload/rpmostreepayload.py:412]: We use `format` for all new code. So `"{} {} exited with code {}".format(cmd, argv, rc)` would be better.

Comment (VladimirSlavik) [inline pyanaconda/payload/rpmostreepayload.py:412]: I'm not 100% sure about the following, but maybe someone else can comment on that.

My own instinct would be to use here something along the lines `"{} failed to create /var/{}".format(cmd, varsubdir)` to make the message more human when it is displayed. Detailed debug output is available in the logs: All executions are logged with complete options, return code and even captured output in `program.log` and also `journal`.

**Review by VladimirSlavik — COMMENTED** [review]
Thanks! Main logic looks good to me.

File: pyanaconda/payload/rpmostreepayload.py (hunk 1 above)
Comment (gicmo) [reply pyanaconda/payload/rpmostreepayload.py:412]: Sure!

Comment (gicmo) [reply pyanaconda/payload/rpmostreepayload.py:412]: Sure, this was just taken from the function it is replacing in the intention to keep the behaviour similar, but I think it makes sense to change it to something more readable.

**Comment by poncovka** [comment]
Thanks!

```
===============================================
   anaconda 33.16.4.12: tests/test-suite.log
===============================================

# TOTAL: 8
# PASS:  8
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0

```

**Comment by VladimirSlavik** [comment]
Looks like the test is waiting for a worker, no idea why.

**Review by rvykydal — APPROVED** [review]
Looks good to me.

**Review by VladimirSlavik — APPROVED** [review]
Oh, I know what I forgot: Looks good to me. Thank you!

**Comment by poncovka** [comment]
Tested with https://bugzilla.redhat.com/show_bug.cgi?id=1935648#c4.

**Comment by poncovka** [comment]
Ported to upstream in https://github.com/rhinstaller/anaconda/pull/3244.

**Pull request merged**